src/deploy-builder.ts

```diff
@@ -0,0 +1,42 @@
+import * as path from 'path'
+import * as fs from 'fs-p'
+
+export class DeployBuilder {
```

Comment:
I think it's great that you're splitting it up into multiple functions, I just think they don't need to live inside a class.